src/lua/luaTask.c

```diff
@@ -124,117 +146,206 @@ size_t get_ontick_freq()
     return 1000 / ticksToMs(onTickSleepInterval);
 }
 
-int getShouldReloadScript(void)
+void* getLua()
 {
-    return g_shouldReloadScript;
+    return g_lua;
 }
 
-void setShouldReloadScript(int reload)
+static void _terminate_lua()
 {
-    g_shouldReloadScript = reload;
+        lua_close(g_lua);
+        g_lua = NULL;
+
+        lua_run_state = LUA_DISABLED;
 }
 
-static void initLuaState()
+void terminate_lua()
 {
-    lockLua();
-    if (g_lua != NULL) lua_close(g_lua);
-    g_lua=lua_newstate( myAlloc, NULL);
-    //open optional libraries
-    luaopen_base(g_lua);
-    luaopen_table(g_lua);
-    luaopen_string(g_lua);
-    luaopen_math(g_lua);
-    registerBaseLuaFunctions(g_lua);
-    registerLuaLoggerBindings(g_lua);
-    unlockLua();
+        lockLua();
+
+        pr_info("lua: Stopping...\r\n");
+
+        if (LUA_DISABLED == lua_run_state)
```

Comment:
can we avoid the use of goto here? We can test if the run state is positive and then terminate lua. It reads more naturally. 

e.g. "If lua is running then terminate it"